src/concrete_solve.jl

```diff
@@ -98,7 +95,10 @@ function DiffEqBase._concrete_solve_adjoint(prob::Union{ODEProblem,SDEProblem},
   else
     has_cb = false
   end
-  default_sensealg = automatic_sensealg_choice(prob,u0,p,has_cb,verbose)
+  default_sensealg = automatic_sensealg_choice(prob,u0,p,verbose)
+  if has_cb
+    default_sensealg = setvjp(default_sensealg, ReverseDiffVJP())

```

Comment:
no tape compilation?

Reply:
yeah.. tape compilation seems to fail for instance in:
```julia
using OrdinaryDiffEq, Zygote
using DiffEqSensitivity, Test, ForwardDiff

abstol = 1e-12
reltol = 1e-12
savingtimes = 0.5

# test the continuous callbacks wrt to the equivalent discrete callback
function f!(du, u, p, t)
  #Bouncing Ball
  du[1] = u[2]
  du[2] = -p[1]
end

# no saving in Callbacks; prescribed vafter and vbefore; loss on the endpoint

tstop = 3.1943828249997
vbefore = -31.30495168499705
vafter = 25.04396134799764

u0 = [50.0, 0.0]
tspan = (0.0, 5.0)
p = [9.8, 0.8]

prob = ODEProblem(f!, u0, tspan, p)

condition2(u, t, integrator) = t == tstop
cb2 = DiscreteCallback(condition2, affect!, save_positions=(false, false))

du01, dp1 = Zygote.gradient(
  (u0, p) -> sum(solve(prob, Tsit5(), u0=u0, p=p,
    callback=cb2,
    tstops=[tstop],
    sensealg=BacksolveAdjoint(autojacvec=ReverseDiffVJP(true)),
    saveat=tspan[2], save_start=false)), u0, p)
# ([-0.0, 1.0], [-9.435743866328323, 0.0])

du01, dp1 = Zygote.gradient(
  (u0, p) -> sum(solve(prob, Tsit5(), u0=u0, p=p,
    callback=cb2,
    tstops=[tstop],
    sensealg=BacksolveAdjoint(autojacvec=ReverseDiffVJP(false)),
    saveat=tspan[2], save_start=false)), u0, p)

# ([1.0, 5.999999999999999], [-17.499999999999964, 0.0]) (correct value)
```

Reply:
oh yes, because the callback machinery has branching.